source/usage-examples/updateMany.txt

```diff
@@ -0,0 +1,113 @@
+===========================
+Update a Multiple Documents
+===========================
+
+.. default-domain:: mongodb
+
+You can update multiple documents using the ``updateMany()`` method on
+a ``MongoCollection`` object. The method accepts a **filter** that matches the
+document you want to update and an **update** statement that instructs the
+driver to change the matching document. The ``updateMany()`` method only
+updates all the documents in the collection that match the filter.
+
+To perform an update with the ``updateMany()`` method, you must pass
+a query filter and an update document. The query filter specifies the criteria
+for which document to perform the update on and the update document provides
+instructions on what changes to make to them.
+
+You can optionally pass an instance of ``UpdateOptions`` to the method in
+order to specify the behavior. For example, if you set the ``upsert`` field of
+the ``UpdateOptions`` object to ``true`` and no documents match the
+specified query filter, the operation inserts a new document composed of
+the fields from both the query and update document. 
+
+Upon successful execution, the ``updateMany()`` method returns an instance
+of ``UpdateResult``. You can retrieve information such as the number of
+documents modified calling the ``getModifiedCount()`` method. If you
+specified ``upsert(true)`` in the ``UpdateOptions`` object and the
+operation results in an insert, you can retrieve the ``_id`` field of the
+new document by calling the ``getUpsertedId()`` method on the
+``UpdateResult``.
+
+If your update operation fails, the driver raises an exception and none of
+the documents matching the filter are updated. For example, if you try to set 
```

Comment:
"none of the documents matching the filter are update"
Is this strictly true? For instance, if you update half the documents, then violate a unique field constraint for a single document halfway through the update, wouldn't some of the documents (the first half) stay updated?

It might be a good idea to avoid mentioning the side-effects of failed operations entirely since that's likely highly dependent on configuration and when the error occurs. Though I'll grant that in the case you provided, none of the documents would update.

Reply:
As far as I've been able to test it, yes. This might be different from "bulkWrite" operations. Will double check with Jeff.